src/pulselistener/pulselistener/lib/mercurial.py

```diff
@@ -72,17 +72,15 @@ def end_of_life(self):
         os.unlink(self.ssh_key_path)
         logger.info('Removed ssh key')
 
-    async def clone(self):
-        # Start by updating the repo in a separate process
-        loop = asyncio.get_running_loop()
-        with ProcessPoolExecutor() as pool:
-            logger.info('Checking out tip in a separate process', repo=self.url)
-            await loop.run_in_executor(
-                pool,
-                batch_checkout,
-                self.url, self.dir, b'tip', self.batch_size,
-            )
-            logger.info('Batch checkout finished')
+    def clone(self):
```

Comment:
Shouldn't we keep cloning in a separate process to avoid blocking other running tasks?

Reply:
This is one step ahead, but when running only the code-review workflow, it's useless to have other processes running before the mercurial checkout is done (except the web process which is in its own process already).

I can rollback to split processes for that part if you want

Reply:
OK, feel free to leave it as it is.